commit/factory.go

```diff
@@ -130,6 +130,15 @@ func (p *PluginFactory) NewReportingPlugin(ctx context.Context, config ocr3types
 			p.lggr,
 			100*time.Millisecond,
 		)
+
+		if err := rmnHomeReader.Ready(); err != nil {
+			return nil, ocr3types.ReportingPluginInfo{}, fmt.Errorf("failed to initialize RMNHome reader: %w", err)
+		}
+		childCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
+		if err := rmnHomeReader.Start(childCtx); err != nil {
```

Comment:
Why not just rmnHomeReader.Start(ctx)? I believe that if the parent context cancels then this child context will cancel even if you have a longer timeout than the parent context.

Reply:
done